### ISSUE TITLE
Not able to upgrade to hashicorp/vault v1.11.3

### ISSUE THREAD

**Issue by skabashnyuk** [hashicorp/vault#17356]
**Describe the bug**
I would like to have a fix for [CVE-2022-40186](https://nvd.nist.gov/vuln/detail/CVE-2022-40186) 
To do that I'm trying to upgrade dependencies of my project to hashicorp/vault v1.11.3. But I met this problem https://github.com/hashicorp/vault/issues/17114.

**To Reproduce**
 - upgrade `hashicorp/vault v1.11.2` ->` hashicorp/vault v1.11.3` 
- `go mod tidy`
- `go build .`
- failed with https://github.com/hashicorp/vault/issues/17114
**Expected behavior**
 - Upgrade should be compatible with the previous version.

**Environment:**
 project https://github.com/redhat-appstudio/service-provider-integration-operator has such a set of dependencies in go.mod

```
	github.com/hashicorp/go-hclog v1.2.2
	github.com/hashicorp/vault v1.11.2
	github.com/hashicorp/vault-plugin-secrets-kv v0.12.1
	github.com/hashicorp/vault/api v1.7.2
	github.com/hashicorp/vault/api/auth/approle v0.1.0
	github.com/hashicorp/vault/api/auth/kubernetes v0.1.0
	github.com/hashicorp/vault/sdk v0.5.3-0.20220721224827-e96a652fbfb0
```

Vault server configuration file(s):

```hcl
n/a
```

**Additional context**
 - https://nvd.nist.gov/vuln/detail/CVE-2022-40186
 -  https://github.com/hashicorp/vault/issues/17114
 - upgrade of `github.com/hashicorp/vault/sdk -> 0.6.0 ` did not helped.

**Comment by cipherboy** [comment]
@skabashnyuk I'm confused by your question. :-) 

Per [comment here](https://github.com/hashicorp/vault/issues/17324#issuecomment-1259684984) by @mpalmi, we do not support pulling _all_ of Vault (`hashicorp/vault`) as a Go module. We only support pulling the API and SDK.

To upgrade your server, I'd recommend pulling binaries from the [releases page](https://releases.hashicorp.com/vault) or upgrading individual API/SDK components when available.

**Comment by hsimon-hashicorp** [comment]
@skabashnyuk Was there a process you followed, or did it just work? I ask for anyone else who may come across this issue in the future. Thanks! :)

**Comment by cipherboy** [comment]
@skabashnyuk Sure, feel free to close. We had some internal discussions that were interested in separating out `NewTestCluster` so it could be ran without having to import `github.com/hashicorp/vault` as an entity, or otherwise developing a better solution for out-of-repo plugin testing &c, but we've not yet gotten to that yet. I suspect we'll probably track it internally and communicate to partners when that is done.

**Comment by cipherboy** [comment]
Yeah, the issue with that in particular is that it is really hard to do: `NewTestCluster` directly interacts with the rest of the `vault` core code, allows most publicly (read: hcl config) accessible hooks to be set, allows modification of some internal values as well (both `/sys` endpoint and other properties)... Plus you directly inject your plugin to the registry by type reference and run the latest Vault core code inside the running test process. 

This makes it rather hard to decouple: you can't just reference the backend if you want to spin up say, a containerized Vault instance against the latest release... You'd have to have a binary path in that case... And containers obviously don't help if you want to test against the latest `main` branch code that hasn't hit an official release yet (to verify your plugin prior to Vault release for instance)... &c.

For the most part, (and speaking mostly to PKI here), the PKI engine has moved to directly initializing and interacting with the backend: 

https://github.com/hashicorp/vault/blob/fdd08482160f3213254e6b8f4c52063382c7323d/builtin/logical/pki/test_helpers.go#L22-L36

This avoids the need to call `NewTestCluster` , as that function is _slow_ (2-4 seconds alone).

If you don't really need a full cluster, I'd look into that as it should be a lot more performant and increase testing.

But obviously some things (like testing PeriodicFunc, invalidation, metrics/telemetry, entity access, or cross-cluster behavior) still require a full Vault cluster and backend-based testing isn't sufficient...
